waiter/src/waiter/statsd.clj

```diff
@@ -348,22 +348,22 @@
               {}
               @aggregation-agent))))
 
-(defn publish-instance-counts-by-metric-group
-  "Publishes a gauge for  the number healthy, unhealthy, and failed
-  instances corresponding to each metric group in the provided map"
-  [counts-by-metric-group]
+(defn publish-metric-group->counts
+  "Publishes a gauge for the number of cpus, mem, healthy instances, unhealthy instances, and
+   failed instances corresponding to each metric group in the provided map."
+  [metric-group->counts]
   (dorun (map (fn [[metric-group {:keys [healthy-instances unhealthy-instances failed-instances cpus mem]}]]
                 (gauge! metric-group "instances.healthy" healthy-instances)
                 (gauge! metric-group "instances.unhealthy" unhealthy-instances)
                 (gauge! metric-group "instances.failed" failed-instances)
                 (gauge! metric-group "cpus" cpus)
                 (gauge! metric-group "mem" mem))
-              counts-by-metric-group)))
+              metric-group->counts)))
 
-(defn process-update-service-instances-message
-  "Merges the current map of resources by metric group with data from a :update-service-instances scheduler message"
-  [counts-by-metric-group {:keys [service-id healthy-instances unhealthy-instances failed-instances] :as message-data}
-   service-id->service-description-fn]
+(defn process-service-instance-state
```

Comment:
Maybe `merge-service-state`? `process` doesn't sound too much like something that takes in a map and returns a map. And the `instance` part seems superfluous...

Reply:
Done.